tests/test-rust-setup.el

```diff
@@ -38,97 +38,147 @@
 (defun crate-file (file-name)
   (expand-file-name file-name "tests/test-crate"))
 
+(defun crate-with-features-file (file-name)
+  (expand-file-name file-name "tests/crate-with-features"))
+
 (defun lib-crate-file (file-name)
   (expand-file-name file-name "tests/custom-lib-target"))
 
 (defun build-script-crate-file (file-name)
   (expand-file-name file-name "tests/build-script-test"))
 
+(defun cdrassoc (sym alist) (cdr (assoc sym alist)))
+
+(defun get-cargo-version ()
+  (let ((cargo (funcall flycheck-executable-find "cargo")))
+    (with-temp-buffer
+      (call-process cargo nil '(t nil) nil "--version")
+      (buffer-string)
+      )))
+
+(defun cargo-version ()
+  (let* ((version-string-parts (split-string (get-cargo-version)))
+         (version (cadr version-string-parts))
+         (version-parts (split-string version "-")))
+    version-parts))
+
+(defun cargo-older-than-1.29-nightly ()
+  (let* ((version-parts (cargo-version))
+         (version (car version-parts))
+         (channel (cadr version-parts)))
+    (if (version< version "1.29")
```

Comment:
More idiomatic would be:

```emacs-lisp
(or (version< version "1.29")
    (and (version= version "1.29") (string= channel "beta")))
```

But wait, the name of the function implies we return `t` when `cargo --version` is more than 1.29-nightly, but we actually do the reverse.  This should be fixed.

Reply:
Function name is `cargo-version-is-older-than-1.29`, so for `1.28.0` it will return `t` (`1.28.0` came out before `1.29.0`, so it's older). Perhaps "older" is a bit confusing?

Reply:
Argh, excuse the brainfart.  `older` is not confusing at all, I was just confused.